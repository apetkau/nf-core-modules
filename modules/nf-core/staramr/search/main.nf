process STARAMR_SEARCH {
    tag "$meta.id"
    label 'process_single'

    conda "bioconda::staramr=0.9.1"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/staramr:0.9.1--pyhdfd78af_0':
        'biocontainers/staramr:0.9.1--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(assembly_fasta) // assembly as fasta file

    output:
    tuple val(meta), path("out/results.xlsx")        , emit: xlsx
    tuple val(meta), path("out/summary.tsv")         , emit: summary_tsv
    tuple val(meta), path("out/detailed_summary.tsv"), emit: detailed_summary_tsv
    tuple val(meta), path("out/resfinder.tsv")       , emit: resfinder_tsv
    tuple val(meta), path("out/plasmidfinder.tsv")   , emit: plasmidfinder_tsv
    tuple val(meta), path("out/mlst.tsv")            , emit: mlst_tsv
    tuple val(meta), path("out/settings.txt")        , emit: txt
    tuple val(meta), path("out/pointfinder.tsv")     , emit: pointfinder_tsv         , optional: true
    tuple val(meta), path("out/hits/resfinder*")     , emit: hits_resfinder_fasta    , optional: true
    tuple val(meta), path("out/hits/pointfinder*")   , emit: hits_pointfinder_fasta  , optional: true
    tuple val(meta), path("out/hits/plasmidfinder*") , emit: hits_plasmidfinder_fasta, optional: true
    path "versions.yml"                              , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    gzip -cdf ${assembly_fasta} > ${assembly_fasta}.uncompressed
    staramr \\
        search \\
        $args \\
        --nprocs $task.cpus \\
        -o out \\
        ${assembly_fasta}.uncompressed

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        staramr : \$(echo \$(staramr --version 2>&1) | sed 's/^.*staramr //' )
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    mkdir out
    touch out/results.xlsx
    touch out/{summary,detailed_summary,resfinder,pointfinder,plasmidfinder,mlst}.tsv
    touch settings.txt
    mkdir out/hits

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        staramr : \$(echo \$(staramr --version 2>&1) | sed 's/^.*staramr //' )
    END_VERSIONS
    """
}
