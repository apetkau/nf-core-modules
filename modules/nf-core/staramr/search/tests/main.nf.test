nextflow_process {

    name "Test Process STARAMR_SEARCH"
    script "../main.nf"
    process "STARAMR_SEARCH"

    tag "modules"
    tag "modules_nfcore"
    tag "staramr"
    tag "staramr/search"

    test("staramr_search - haemophilus_influenzae - genome_fna_gz") {
        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data['haemophilus_influenzae']['genome']['genome_fna_gz'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },

                // md5 for these outputs change on each run (they contain dates) and so I can only compare filename
                { assert snapshot(file(process.out.results_xlsx.get(0).get(1)).name).match("test_results.xlsx") },
                { assert snapshot(file(process.out.settings_txt.get(0).get(1)).name).match("test_settings.txt") },

                { assert snapshot(path(process.out.summary_tsv.get(0).get(1))).match("test_summary.tsv") },
                { assert snapshot(path(process.out.detailed_summary_tsv.get(0).get(1))).match("test_detailed_summary.tsv") },
                { assert snapshot(path(process.out.resfinder_tsv.get(0).get(1))).match("test_resfinder.tsv") },
                { assert snapshot(path(process.out.plasmidfinder_tsv.get(0).get(1))).match("test_plasmidfinder.tsv") },
                { assert snapshot(path(process.out.mlst_tsv.get(0).get(1))).match("test_mlst.tsv") }
            )
        }
    }
}
