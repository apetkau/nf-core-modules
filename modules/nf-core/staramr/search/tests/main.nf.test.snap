{
    "test_summary_tsv": {
        "content": [
            "summary.tsv.gz:md5,45a667d8ad6fff574684a8a55c938204"
        ],
        "timestamp": "2023-11-28T16:52:00.869645524"
    },
    "test_plasmidfinder_tsv": {
        "content": [
            "plasmidfinder.tsv.gz:md5,4b4cf344b708fd3f7f2517ed92318f19"
        ],
        "timestamp": "2023-11-28T15:43:08.308529861"
    },
    "test_settings_txt": {
        "content": [
            "settings.txt.gz"
        ],
        "timestamp": "2023-11-28T15:43:08.29237937"
    },
    "test_detailed_summary_tsv": {
        "content": [
            "detailed_summary.tsv.gz:md5,3bdb116ef6b6150c4f03649a4e7a15ec"
        ],
        "timestamp": "2023-11-28T16:52:00.899729415"
    },
    "test_resfinder_tsv": {
        "content": [
            "resfinder.tsv.gz:md5,90091a6d9d61d3303cb47823dbb6db3d"
        ],
        "timestamp": "2023-11-28T16:52:00.930882269"
    },
    "test_mlst_tsv": {
        "content": [
            "mlst.tsv.gz:md5,8f95c8e181820d1bbf46d44d49f0f6cf"
        ],
        "timestamp": "2023-11-28T16:52:00.962510157"
    },
    "test_results_xlsx": {
        "content": [
            "results.xlsx"
        ],
        "timestamp": "2023-11-28T15:43:08.2814982"
    }
}